engine/runtime-language-epb/src/main/java/org/enso/interpreter/epb/node/ForeignEvalNode.java

```diff
@@ -106,86 +98,44 @@ private void lockAndParse() throws IllegalStateException {
 
   private void parseJs() {
     EpbContext context = EpbContext.get(this);
-    GuardedTruffleContext outer = context.getCurrentContext();
-    GuardedTruffleContext inner = context.getInnerContext();
```

Comment:
@Akirathan, you removed this code in 29d7963af518a13ba055d66fa952894232012b4d, but how that's supposed to work? JavaScript probably needs its own context unless we want to make Enso single-threaded-only!

We might fix the `RuntimeVisualizationTest` to be single threaded (and then JavaScript invocation from the same context could work), but that'd be just a test. As soon as we allow multi-threaded Enso execution somewhere, we'd be in the same problem - `JavaScriptLanguage` refusing to start in multiple threads context.

Reply:
I think the inner context is still needed. Returning it back in https://github.com/enso-org/enso/pull/7882/commits/6ce3a1b0b2dd5d0984b17480717e9a22447426be